components/Feature/VulnerabilitiesGrid/index.js

```diff
@@ -42,42 +43,75 @@ const VulnerabilitiesGrid = ({ onUpdate }) => {
+                  const cbId = `${id}-v-${labelToId(label)}`;
                   return (
-                    <Checkbox
-                      key={`${id}-v-${j}`}
-                      label={text}
-                      name={`${id}-v-${j}`}
-                      onClick={() =>
-                        updateVulnerabilities(`${id}-v-${j}`, text)
-                      }
-                    />
+                    <React.Fragment key={cbId}>
+                      <Checkbox
+                        label={label}
+                        name={cbId}
+                        onClick={() => updateVulnerabilities(cbId, label)}
+                      />
+                      {textinputs &&
+                        grid.vulnerabilities[cbId] &&
+                        textinputs.map(({ label }) => {
+                          const inputId = `${cbId}-${labelToId(label)}-i`;
+                          return (
+                            <TextInput
+                              name={inputId}
+                              key={inputId}
+                              label={label}
+                              onChange={val =>
+                                updateVulnerabilities(inputId, val)
+                              }
+                            />
+                          );
+                        })}
+                    </React.Fragment>
                   );
                 })}
               </CheckboxList>
             </div>
             <div className="govuk-grid-column-one-half">
               <CheckboxList className="asset">
-                {assets.map(({ text }, j) => {
+                {assets.map(({ label, textinputs }) => {
+                  const cbId = `${id}-a-${labelToId(label)}`;
                   return (
-                    <Checkbox
-                      key={`${id}-a-${j}`}
-                      label={text}
-                      name={`${id}-a-${j}`}
-                      onClick={() => updateAssets(`${id}-a-${j}`, text)}
-                    />
+                    <React.Fragment key={cbId}>
+                      <Checkbox
+                        key={cbId}
+                        label={label}
+                        name={cbId}
+                        onClick={() => updateAssets(cbId, label)}
+                      />
+                      {textinputs &&
+                        grid.assets[cbId] &&
+                        textinputs.map(({ label }) => {
+                          const inputId = `${cbId}-${labelToId(label)}-i`;
+                          return (
+                            <TextInput
+                              name={inputId}
+                              key={inputId}
+                              label={label}
+                              onChange={val => updateAssets(inputId, val)}
+                            />
+                          );
+                        })}
+                    </React.Fragment>
```

Comment:
I see this used in two places, is it worth pulling it out into a separate component - something like `CheckboxWithInput`?

I'm thinking it could be used something like this (of course we'd have to pass through a bunch more things)...

```
<CheckboxWithInput
  checked={grid.assets[cbId]}
  onChange={updateAssets}
  inputs={textinputs} />
```

What do you think?

Reply:
Right now we're only using this only in this scenario. We can think about extracting it in a separate component later on in case we need.